src/geo/locate.js

```diff
@@ -40,6 +44,68 @@ const handleQuery = async (opts) => {
   }
 }
 
+const locateCity = async ({ city, region, country }) => {
+  const query = {
+    text: `${city}, ${region} ${country}`,
+    size: 1
+  }
+
+  // check if cache has it first
+  const lruKey = JSON.stringify(query)
+  if (lru.has(lruKey)) return lru.get(lruKey)
```

Comment:
You'll want to do 3 LRU caches here - one for intersection lookup, one for the city lookup, and one for the address lookup.